docs/dev/testing.md

```diff
@@ -0,0 +1,76 @@
+# Assisted Installer Testing
+
+The assisted-installer tests are divided into 3 categories:
+
+* **Unit tests** - Focused on a module/function level while other modules are mocked.
+Unit tests are located next to a module named `module_name_test.go`
+
+* **Subsystem tests** - Focused on the component while mocking other component. (e.g. agent/installer results).
```

Comment:
not sure I understand "agent/installer results"
what do you mean?

Reply:
Examples for mocks. I'll rephrase